File: src/main/java/com/machfour/macros/objects/FoodAttribute.java
package com.machfour.macros.objects;

import com.machfour.macros.core.*;

public class FoodAttribute extends MacrosEntity<FoodAttribute> {

    private FoodAttribute(ColumnData<FoodAttribute> data, ObjectSource objectSource) {
        super(data, objectSource);
    }

    @Override
    public Table<FoodAttribute> getTable() {
        return Schema.FoodAttributeTable.instance();
    }

    public static Factory<FoodAttribute> factory() {
        return FoodAttribute::new;
    }
    @Override
    public Factory<FoodAttribute> getFactory() {
        return factory();
    }

}

File: src/main/kotlin/com/machfour/macros/objects/FoodAttribute.kt
package com.machfour.macros.objects

import com.machfour.macros.core.*

class FoodAttribute private constructor(data: ColumnData<FoodAttribute>, objectSource: ObjectSource)
    : MacrosEntityImpl<FoodAttribute>(data, objectSource) {

    companion object {
        val table: Table<FoodAttribute>
            get() = Schema.FoodAttributeTable.instance
        val factory: Factory<FoodAttribute> = Factory { dataMap, objectSource -> FoodAttribute(dataMap, objectSource) }
    }

    override val factory: Factory<FoodAttribute>
        get() = Companion.factory
    override val table: Table<FoodAttribute>
        get() = Companion.table



}
